pkg/operator/operator.go

```diff
@@ -212,32 +213,6 @@ func (c *csiDriverOperator) sync() error {
 		return err
 	}
 
-	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
-		// CR is NOT being deleted, so make sure it has the finalizer
-		if addFinalizer(instance, operatorFinalizer) {
-			c.client.UpdateFinalizers(instance)
-		}
-	} else {
-		// User tried to delete the CR, let's evaluate if we can
-		// delete the operand and remove the finalizer from the CR
-		ok, err := c.isCSIDriverInUse()
-		if err != nil {
-			return err
-		}
-
-		if ok {
-			// CSI driver is being used but CR has been marked for deletion,
-			// so we can't delete or sync the operand nor remove the finalizer
-			klog.Warningf("CR is being deleted but there are resources using the CSI driver")
-			return nil
-		} else {
-			// CSI driver is not being used, we can go ahead and remove finalizer and delete the operand
-			removeFinalizer(instance, operatorFinalizer)
-			c.client.UpdateFinalizers(instance)
-			return c.deleteAll()
-		}
-	}
-
```

Comment:
Here we were mutating the input instead of the copy of it (created on line 246 below).